include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -1039,16 +1050,18 @@ def StructInjectOp : MooreOp<"struct_inject"> {
     ```
     See IEEE 1800-2017 § 7.2. "Assigning to structures".
   }];
-  let arguments = (ins UnpackedType:$LHS, StrAttr:$memberName,
-  UnpackedType:$RHS);
+  let arguments = (ins UnpackedType:$input, StrAttr:$fieldName,
+  UnpackedType:$newValue);
   let results = (outs UnpackedType:$result);
```

Comment:
Can you make `$input` and `$result` an `AnyStructType`? And maybe also add a `TypesMatch<...>` constraint between `$input` and `$result`. (The result struct has the same type as the input struct.) That again lets you remove the `-> type($result)` part of the assembly format.

Reply:
Since we already have a verifier that verify the type of inputs, we do not need to have this exact type declartion. Types like anyStructType could be replaced to unpacked type.